output/report.go

```diff
@@ -44,7 +44,9 @@ func GetReporter(names []string) (Reporter, error) {
 	for _, name := range names {
 		switch name {
 		case "list":
-			mux = append(mux, List)
+			mux = append(mux, &List{Compact: false})
+		case "compact":
+			mux = append(mux, &List{Compact: true})
```

Comment:
We should optimize for:

- Building new reporters - if a third reporter wants to come in and use some of these pieces, the example this sets fourth is to come into `List` and make it do even more. A common lib would be better.
- Single responsibility - a reporter should do 1 thing well, and that thing is show one version of output. A flag that changes its behavior breaks this idea.
- Isolation - a new reporter shouldn't change another one. In this case, instead of just a new reporter, we're changing `List` significantly.